float64[] positions
float64[] torques # optional feedforward torques
float64 time
